### ISSUE TITLE
Rationalize API

### ISSUE THREAD

**Issue by wsargent** [playframework/play-ws#101]
See https://github.com/playframework/playframework/issues/2240 and https://github.com/playframework/playframework/issues/5813

play.mvc.Http.RequestHeader.headers() returns a Map<String, String[]>
play.mvc.Http.RequestHeader.getHeader() return just a String and not String[]
WS.Response.getHeader() return just a String and not a String[]

**Comment by jjpatel361** [comment]
@wsargent I would be willing to submit a PR on this given the guidance from you. I have seen the discussion thread. I am wondering how is this different than having 
```java
WSResponse wt = request.setHeader("Authorization", API_KEY).get().toCompletableFuture().get();
wt.getAllHeaders() // returns Map<String, List<String>>
```

Is this already taken?

**Comment by marcospereira** [comment]
See #88 too.

**Comment by jjpatel361** [comment]
Ok so there will be three methods  
```java
 wt.getAllHeaders() // returns Map<String, List<String>>
 wt.getHeader(arg0) // returns String value from the map
// To be added should return one header, 
//if its a Collection then 1st element of the collection
 wt.getHeader() 
```
That's my understanding of it.

**Comment by jjpatel361** [comment]
Is my understanding correct on above part? or I am still missing something.

**Comment by marcospereira** [comment]
This was closed in #115.
